caching openmpi, .julia, rm Simulation

jwscook opened the pull request:
This might well speed up travis, but will have to run travis twice to find out.

Commit:
caching openmpi, .julia, rm Simulation

SCRC-87

jwscook committed:
InstallMPI.sh openmpi cached to wrong place

SCRC-87 

See failure https://travis-ci.org/github/ScottishCovidResponse/Simulation.jl/jobs/683003127
Julia tries to "Activating new environment at `~/build/ScottishCovidResponse/Simulation.jl/openmpi/Project.toml`" - openmpi is in the wrong place for caching

jwscook committed:
.travis.yml update cache location

SCRC-87

Should have gone in with previous commit to this branch

jwscook committed:
InstallMPI.sh linux build had wrong JULIA_MPI_PATH

SCRC-87

ERROR: LoadError: LoadError: InitError: could not load library "/home/travis/build/ScottishCovidResponse/Simulation.jl/openmpi/lib/libmpi"
which isn't where openmpi should be - should be in ~

Commit:
cache openmpi, format installMPI and fix paths

SCRC-87

Commit:
faster macosx build, test JULIA_MPI_PATH

SCRC-87
print statements at top of test/runtests.jl that need to be removed eventually
print statements in .travis.yml ""

Commit:
changed back runtests.jl, travis.yml & installMPI

SCRC-87 caching openmpi dir, and moving it into the JULIA_PROJECT path at the start of the travis script

Commit:
travis.yml & installMPI copying openmpi

SCRC-87 caching openmpi dir, and moving openmpi dir to cache

Commit:
fixes to travis.yml and InstallMPI.sh

SCRC-87

Commit:
stab in the dark in to travis.yml and InstallMPI.sh

SCRC-87

Commit:
old cache messing things up travis.yml and InstallMPI.sh

SCRC-87

Commit:
getting things wrong InstallMPI.sh

SCRC-87

Commit:
got error MPI error in test

SCRC-87 ERROR: LoadError: LoadError: InitError: MPI library has changed, please restart Julia. was copying mpi stuff into Simulation.jl, now setting JULIA_MPI_PATH

Commit:
installMPI.sh error - wasnt checking cache right

SCRC-87 wasnt checking in right palce for mpirun

Commit:
rm Simulation.jl in before_cache:

SCRC-87 was only deleting the .julia/packages/Simulation.jl after_success and not after_success. now done in before_cache:. Can't do it in after_script because that happens after cache:.

Commit:
wrong path installMPI.sh for osx

SCRC-87 had ./openmpi not ~/openmpi

Commit:
not caching on osx

SCRC-87 good old print statements

Commit:
working in osx mpi cache

SCRC-87

Commit:
path not evaluated correctly maybe on osx mpi cache

SCRC-87

Commit:
needed symlink after all osx installmpi

SCRC-87

Commit:
added osx ENV vars for mpi cache

SCRC-87

Commit:
didn't reallise that brew symlinks installs

SCRC-87

Commit:
Cellar wasn't cached on osx

SCRC-87

Commit:
printing out /usr/local cellar etc

SCRC-87 working on installMPI.sh for macosx, trying to get it to recognise the cached directory /usr/local/Cellar

Commit:
printing out /usr/local cellar etc

SCRC-87 working on installMPI.sh for macosx, trying to get it to recognise the cached directory /usr/local/Cellar

Commit:
got the path wrong osx mpi cache

SCRC-87 working on installMPI.sh for macosx, got wrong path to mpi installation under Cellar

Commit:
osx mpi cache set env vars

SCRC-87 need to set PATH and LD_LIBRARY_PATH

Commit:
osx mpi cache DYLD_LIBRARY_PATH

SCRC-87 set DYLD_LIBRARY_PATH not LD_LIBRARY_PATH

Commit:
relink brew installed cached libs

SCRC-87 might need to manually set DYLD_LIBRARY_PATH now attempting to relink with brew

claireh93 reviewed (commented):
Hi James, I have lost track a bit, is this ready to be merged?

Commit:
relink too slow - last chance

SCRC-87 trying different install path so maybe fewer symlinks to deal with?

jwscook commented:
Not ready yet I'm afraid :(

Commit:
last attempt at caching MPI on macosx

SCRC-87

Commit:
print statement on macosx

SCRC-87

Commit:
wrong path

SCRC-87

Commit:
weirdly directory not there

SCRC-87 travis says /Users/travis/openmpi is brought out of cache, yet ls reveals nothing

claireh93 commented:
No worries! Sorry I was just going through open PRs last night, but now with my brain switched on I see you need to run travis twice on these commits

Commit:
revert to unlink-link solution

SCRC-87 speeding up unlink link by passing hw.ncpu to xargs via -P argument

Commit:
libmpi not on /usr/local/Cellar anymore

SCRC-87 need to rebuild cellar from scratch, then use a new cached version

Commit:
try again now Cellar is cached

SCRC-87 use cached cellar, repair symlinks

Commit:
moving export JULIA_MPI_PATH into installMPI.sh

SCRC-87 it was in .travis.yml but makes more sense in installMPI.sh
